src/modules/powerrename/lib/PowerRenameRegEx.cpp

```diff
@@ -194,73 +226,90 @@ HRESULT CPowerRenameRegEx::Replace(_In_ PCWSTR source, _Outptr_ PWSTR* result)
     *result = nullptr;
 
     CSRWSharedAutoLock lock(&m_lock);
-    HRESULT hr = (source && wcslen(source) > 0 && m_searchTerm && wcslen(m_searchTerm) > 0) ? S_OK : E_INVALIDARG;
-    if (SUCCEEDED(hr))
+    HRESULT hr = S_OK;
+    if (!(m_searchTerm && wcslen(m_searchTerm) > 0 && source && wcslen(source) > 0))
     {
-        wstring res = source;
-        try
+        return hr;
```

Comment:
We are returning S_OK, while before we were returning E_INVALIDARG.

Reply:
Search term or resource can be emty string. (No entry in the search for area or extension only flag is selected when there is no extension.) I think method should just do nothing in this case without failing. Return values of this method was not checked before and regexWorkerThread was proceeding despite of E_INVALIDARG.

Reply:
OK